### DISCUSSION TITLE
Share graphql query between scenarios

### DISCUSSION THREAD

**Discussion by moduval** [artilleryio/artillery#1652] [Artillery]
Hello,

I have multiple scenarios querying the same GraphQL query. Since a GraphQL query can be big depending on the field I request and can change over time, it would be nice if we could put our query in a file that we can share between scenarios.

This way if the query change, we would have to change it only on one file instead of going through every scenario.

**Comment by juanig1** [comment]
Hey!

There is no GraphQL specific behavior for that in artillery but you could use a custom js helper like [this](https://github.com/artilleryio/artillery/tree/master/examples/generating-vu-tokens) to return your query (or parts of it), and reuse that.
Edit: depending on your query, [variables](https://www.artillery.io/docs/guides/guides/test-script-reference#variables---inline-variables) can also be helpful
